src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java

```diff
@@ -315,4 +337,30 @@ private void logStepNotInEntityScreenError(EntityScreen entityScreen, SessionDat
         neededDatum.getDataId(), step.getValue(), nodeSetString, referencesString));
 
     }
+
+    /**
+     * Execute the post request associated with the sync screen and perform a sync if necessary.
+     */
+    public void doPostAndSync(MenuSession menuSession, FormplayerSyncScreen screen) throws SyncRestoreException {
```

Comment:
nit: A Factory class doesn't seem like a great place for this public method, think it would be better to put it in it's own helper class if possible.

Reply:
that makes sense, [f817923](https://github.com/dimagi/formplayer/pull/1622/commits/f817923ccd4b5aa1fa100636d70b2035571aaa2a)